casper/src/main/scala/coop/rchain/casper/BlockDag.scala

```diff
@@ -0,0 +1,31 @@
+package coop.rchain.casper
+
+import com.google.protobuf.ByteString
+import coop.rchain.casper.Estimator.{BlockHash, Validator}
+import coop.rchain.casper.protocol.BlockMessage
+
+import scala.collection.immutable.{HashMap, HashSet}
+
+final case class BlockDag(idToBlocks: HashMap[Int, BlockMessage],
+                          blockLookup: HashMap[ByteString, BlockMessage],
+                          childMap: HashMap[BlockHash, HashSet[BlockHash]],
+                          latestMessages: HashMap[Validator, BlockHash],
+                          currentId: Int)
+
+object BlockDag {
+  def apply(idToBlocks: HashMap[Int, BlockMessage],
```

Comment:
This isn't necessary to provide explicitly because it would be auto-generated by the compiler since `BlockDag` is a case class.